rs/boundary_node/rate_limits/api/src/schema_versions/v1.rs

```diff
@@ -209,13 +296,67 @@ mod test {
                 subnet_id: None,
                 methods_regex: Some(Regex::new("^.*$").unwrap()),
                 request_types: None,
+                ip: Some(IpNet::new_assert(
+                    IpAddr::V4(Ipv4Addr::new(10, 1, 1, 0)),
+                    24
+                )),
+                ip_prefix_group: Some(IpPrefixes { v4: 24, v6: 64 }),
                 limit: Action::Limit(100, Duration::from_secs(1)),
             }
         );
 
+        // Bad prefix lengths
+        let rule_raw = indoc! {"
+        canister_id: aaaaa-aa
+        methods_regex: ^.*$
+        ip: 10.1.1.0/24
+        ip_prefix_group:
+          v4: 33
+          v6: 64
+        limit: 100/1s
+        "};
+
+        assert!(RateLimitRule::from_bytes_yaml(rule_raw.as_bytes()).is_err());
+
+        let rule_raw = indoc! {"
+        canister_id: aaaaa-aa
+        methods_regex: ^.*$
+        ip: 10.1.1.0/24
+        ip_prefix_group:
+          v4: 24
+          v6: 129
+        limit: 100/1s
+        "};
+
+        assert!(RateLimitRule::from_bytes_yaml(rule_raw.as_bytes()).is_err());
+
+        // limit: block with ip prefixes
+        let rule_raw = indoc! {"
+        canister_id: aaaaa-aa
+        methods_regex: ^.*$
+        ip: 10.1.1.0/24
+        ip_prefix_group:
+          v4: 24
+          v6: 64
+        limit: block
+        "};
+
+        assert!(RateLimitRule::from_bytes_yaml(rule_raw.as_bytes()).is_err());
+
+        // No conditions
+        let rule_raw = indoc! {"
+        ip_prefix_group:
+            v4: 33
```

Comment:
```suggestion
            v4: 24
```
otherwise, it is not clear whether the error comes becomes of the bad prefix length or the missing conditions.

Reply:
Yep thanks, copy paste is like this :)